calico/felix/ipsets.py

```diff
@@ -159,103 +176,137 @@ def on_tags_update(self, profile_id, tags):
         else:
             self.tags_by_prof_id[profile_id] = tags
 
-    def _process_tag_updates(self, profile_id, old_tags, new_tags):
-        """
-        Updates the active ipsets associated with the change in tags
-        of the given profile ID.
-        """
-        endpoint_ids = self.endpoint_ids_by_profile_id.get(profile_id, set())
-        _log.debug("Endpoint IDs with this profile: %s", endpoint_ids)
-        added_tags = new_tags - old_tags
-        _log.debug("Profile %s added tags: %s", profile_id, added_tags)
-        removed_tags = old_tags - new_tags
-        _log.debug("Profile %s removed tags: %s", profile_id, removed_tags)
-        for added, upd_tags in [(True, added_tags), (False, removed_tags)]:
-            for tag in upd_tags:
-                if added:
-                    self.endpoint_ids_by_tag[tag] |= endpoint_ids
-                else:
-                    self.endpoint_ids_by_tag[tag] -= endpoint_ids
-                if self._is_starting_or_live(tag):
-                    # Tag is in-use, update its members.
-                    ipset = self.objects_by_id[tag]
-                    for endpoint_id in endpoint_ids:
-                        endpoint = self.endpoints_by_ep_id[endpoint_id]
-                        for ip in map(futils.net_to_ip,
-                                      endpoint.get(self.nets_key, [])):
-                            if added:
-                                ipset.add_member(ip, async=True)
-                            else:
-                                ipset.remove_member(ip, async=True)
+    def _extract_ips(self, endpoint):
+        if endpoint is None:
+            return set()
+        return set(map(futils.net_to_ip,
+                       endpoint.get(self.nets_key, [])))
+
 
     @actor_message()
     def on_endpoint_update(self, endpoint_id, endpoint):
+        """
+        Update tag memberships and indexes with the new endpoint dict.
```

Comment:
Perhaps a bit pernickety, but "indexes" -> "indices" throughout.

"Indexes" appears to be correct in the context of the index of a book, but not in the context of indexed data.